### PR TITLE
Add WriteModelBinary trait + impl.

### PR DESCRIPTION
**Pull request by danieldk**

### CODE REVIEW

File: finalfrontier/src/model.rs [hunk 1]
```diff
@@ -101,6 +104,40 @@ impl TrainModel where {
...
+        write.write_u32::<LittleEndian>(self.config.buckets_exp)?;
+        write.write_u32::<LittleEndian>(self.config.negative_samples)?;
+        write.write_f32::<LittleEndian>(self.config.lr)?;
+        write.write_u64::<LittleEndian>(self.vocab.n_tokens() as u64)?;
+        write.write_u64::<LittleEndian>(self.vocab.len() as u64)?;
+
+        for token in self.vocab.types() {
+            write.write_u32::<LittleEndian>(token.token().len() as u32)?;
+            write.write_all(token.token().as_bytes())?;
+            write.write_u32::<LittleEndian>(token.count() as u32)?;
```
Comment (danieldk) [inline finalfrontier/src/model.rs]: I will amend this commit, with ccrawl-sized corpora, we are getting dangerously close to the possibility that tokens occur more than 2^32 times.

File: finalfrontier/src/io.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
-use std::io::{self, BufRead, Lines};
+use std::io::{self, BufRead, Lines, Read, Write};
```
Comment (DiveFish) [inline finalfrontier/src/io.rs]: Unused import `Read`- will it be used later?

File: finalfrontier/src/model.rs [hunk 3]
```diff
@@ -101,6 +104,40 @@ impl TrainModel where {
     }
 }
 
+impl<W> WriteModelBinary<W> for TrainModel
+where
+    W: Write,
+{
+    fn write_model_binary(&self, write: &mut W) -> Result<(), Error> {
+        write.write_all(&[b'D', b'F', b'F'])?;
+        write.write_u32::<LittleEndian>(1)?;
```
Comment (DiveFish) [inline finalfrontier/src/model.rs]: Not printing `config.model`?

File: finalfrontier/src/io.rs (hunk 2 above)
Comment (danieldk) [reply finalfrontier/src/io.rs]: Indeed :). There is a trait to load models as well.

File: finalfrontier/src/model.rs (hunk 3 above)
Comment (danieldk) [reply finalfrontier/src/model.rs]: Good catch, the loss is also missing :).

**Pull request merged**